[WD-20743] feat: new /data/lakehouse page

muhammad-ali-pk opened the pull request:
~**NOTE**: Replace `formId` in `/data/form-data.json` once marketo team provides it. Currently using another id from a similar form.~

## Done

- Built a new page i.e., /data/lakehouse
- Add entry to data/sitemap.xml
- Added modal form

## QA

- View the site in your web browser at: https://canonical-com-1658.demos.haus/data/lakehouse
  - Make sure to test on all screen sizes i.e., mobile, tablet and desktop.
- Verify the page design matches [figma](https://www.figma.com/design/uJUwbUAGDog4KCMivrMpnl/canonical.com-data?node-id=3626-4237&t=4klvTIOH0AJbjFmU-0)
- Verify the page content matches [copydoc](https://docs.google.com/document/d/1iG9Fcg1w9FWyWO84Gj6RFFYdu5xzEDFBMxnjAAQFW34)

## Issue / Card

Fixes #[WD-20743](https://warthogs.atlassian.net/browse/WD-20743)

[WD-20743]: https://warthogs.atlassian.net/browse/WD-20743?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://canonical-com-1658.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/12279)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

mattea-turic commented:
Thank youuu @muhammad-ali-pk !

Feedback:
---
**For "Data lakehouse architecture":**
- [ ] The image should be sized so that e.g. "ingestion and storage layer" starts at 50% and aligns with the text above and below. Not sure what's going wrong as you've copied it exactly as in Figma (using cinematic ar, starting at 5th column, etc.), so I'm a little lost and will do some further investigation

**For "Data lakehouse vs. data lake...":**
- [ ] Could you add a `<br>` after "data lake" pls

And thanks @eliman11 for bringing up the HRs!

muhammad-ali-pk commented:
@eliman11 @mattea-turic  Thanks! I have addressed your comments.

@eliman11 PS., just so you know, the `.p-table--mobile-card` vanilla class, out of the box, applies card styles to smaller screens (both mobile and tablets), so the table is only going to be visible on large screen.

@mattea-turic It could be an issue with the image itself? Not sure either. I tried playing around with grid and image's ar but couldn't make it aligned.

eliman11 commented:
Thanks @muhammad-ali-pk! Noted on the cards, thanks for letting me know! There's just this one comment on hiding the charts that needs to be addressed - it seems like the image is still showing for me on mobile.
> **For Data lakehouse architecture**
> 
> * [x]  Could we hide the chart image for screens smaller than 500px in width? Beyond this size the text in the chart becomes difficult to read.

muhammad-ali-pk commented:
@eliman11 Sorry about that. Should be hidden on smaller screens now.
Thanks!

eliman11 commented:
No worries! Perfect, thanks for the quick change :)

mattea-turic commented:
Adding design +1 while I continue trying to figure out the diagram :')
As discussed on MM @muhammad-ali-pk

britneywwc commented on templates/data/sitemap.xml:
```diff
@@ -64,4 +64,8 @@
     <loc>https://canonical.com/data/streaming</loc>
     <changefreq>monthly</changefreq>
   </url>
+    <url>
```
```suggestion
  <url>
```

britneywwc commented on templates/data/lakehouse/index.html:
```diff
@@ -0,0 +1,426 @@
+                    Apache®, Apache Kafka, Kafka®, and the Kafka logo are either registered trademarks or trademarks of the Apache Software Foundation in the United States and/or other countries.
+                </p>
+                <p class="u-text--muted">
+                    Apache®, Apache Spark, Spark®, and the Spark logo are either registered trademarks or trademarks of the Apache Software Foundation in the United States and/or other countries.
```
We should use HTML entities to replace the symbols ®

britneywwc commented on templates/data/lakehouse/index.html:
```diff
@@ -0,0 +1,426 @@
+    </section>
+
+    <section class="p-section">
+        <hr class="p-rule is-fixed-width " />
```
```suggestion
        <hr class="p-rule is-fixed-width" />
```

britneywwc commented on templates/data/lakehouse/index.html:
```diff
@@ -0,0 +1,426 @@
+                    </div>
+                </div>
+
+                <div class="row">
```
The rows should be wrapped in a shallow section as per design

britneywwc commented on templates/data/lakehouse/index.html:
```diff
@@ -0,0 +1,426 @@
+                    </div>
+                </div>
+            </div>
+            <div class="p-section--shallow">
```
There is no shallow section wrapping over the entire table here

britneywwc commented on templates/data/lakehouse/index.html:
```diff
@@ -118,54 +118,66 @@ <h2>Data lakehouse architecture</h2>
+                <div class="p-section--shallow">
+                    <div class="row">
+                        <div class="col-9 col-start-large-4">
+                            <hr class="is-muted" />
```
```suggestion
                            <hr class="p-rule--muted" />
```
We should update the muted class to the latest convention

muhammad-ali-pk replied:
That's straight from Vanilla pattern's generated HTML for tiered list (also the missing shallows you mentioned earlier). I have a [PR](https://github.com/canonical/vanilla-framework/pull/5503) up there to add an image feature to tiered list, so this entire section is going to be replaced by a pattern macro.

britneywwc reviewed (approved):
Looks amazing, thanks!
